src/sql/workbench/api/browser/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -709,4 +712,12 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 			}
 		});
 	}
+
+	$createNotebookDocument(providerId: string, contents: azdata.nb.INotebookContents): Promise<azdata.nb.NotebookDocument> {
+		return this._commandService.executeCommand(NewNotebookAction.INTERNAL_NEW_NOTEBOOK_CMD_ID, {
```

Comment:
Any reason you're not using the Notebook service here? Not having typing on the executeCommand means this could silently break if we change what the command returns.

Reply:
The notebook service's openNotebook method requires a URI in order to open the notebook, which we don't have here. The New Notebook command builds an untitled URI before calling showNotebookDocument with the provided options.

Reply:
I could also take a look at moving the New Notebook command logic into the notebook service later, since it's still weird that we contribute some core notebook commands from an extension.

Reply:
Sounds good, this could definitely use some improvement. Anyways fine to do as a follow up task since we do this kind of stuff in other places already anyways.